idl-compiler.py

```diff
@@ -1434,6 +1443,18 @@ def add_view(cout, cls):
             }}
         """).format(f=DESERIALIZER, **locals()))
 
+        if is_vector(m.type):
+            item_type = basic_type(m.type)
+            fprintln(cout, reindent(4, """
+                auto {name}_range() const {{
+                  return seastar::with_serialized_stream(v, [this] (auto& v) {{
+                   auto in = v;
+                   {skip}
+                   return range_deserializer<{item_type}>(in);
+                  }});
+                }}
+            """).format(f=DESERIALIZER, **locals()))
+
```

Comment:
Why not replace the existing function?

Reply:
Need unit test for the new function.

Reply:
> Why not replace the existing function?

It's possible.  I was concerned it could hurt performance, but if the std::optional<T> is moved to the range_deserializer rather than in the iterator that shouldn't be a problem.

> Need unit test for the new function.

`test_writing_and_reading_gently` tests it.
I'll add that to the commitlog.